app/internal/import_file.py

```diff
@@ -0,0 +1,198 @@
+from collections import defaultdict
+import datetime
+import os
+from pathlib import Path
+import re
+from typing import Any, Dict, List, Tuple, Union
+
+from icalendar import Calendar
+
+from app.config import (
+    EVENT_CONTENT_LIMIT,
+    EVENT_HEADER_LIMIT,
+    EVENT_HEADER_NOT_EMPTY,
+    MAX_EVENTS_START_DATE,
+    MAX_FILE_SIZE_MB,
+    NUM_OF_VALUES,
+    VALID_FILE_EXTENSION,
+    VALID_YEARS
+)
+from app.database.models import Event
+from app.database.database import SessionLocal
+
+
+DATE_FORMAT = "%m-%d-%Y"
+DESC_EVENT = "VEVENT"
+EVENT_PATTERN = re.compile(r"^(\w{" + str(EVENT_HEADER_NOT_EMPTY) + "," +
+                           str(EVENT_HEADER_LIMIT) + r"}),\s(\w{0," +
+                           str(EVENT_CONTENT_LIMIT) +
+                           r"}),\s(\d{2}-\d{2}-\d{4})," +
+                           r"\s(\d{2}-\d{2}-\d{4})$")
+
+
+def is_file_size_valid(file: str, max_size: int = MAX_FILE_SIZE_MB) -> bool:
+    file_size = os.stat(file).st_size / 1048576  # convert bytes to MB.
+    return file_size <= max_size
+
+
+def is_file_extension_valid(file: str,
+                            extension: Union[str, Tuple[str, ...]]
+                            = VALID_FILE_EXTENSION) -> bool:
+    return file.lower().endswith(extension)
+
+
+def is_file_exist(file: str) -> bool:
+    return Path(file).is_file()
+
+
+def is_date_in_range(date: Union[str, datetime.datetime],
+                     valid_dates: int = VALID_YEARS) -> bool:
+    """
+    check if date is valid and in the range according to the rule we have set
+    """
+    now_year = datetime.datetime.now().year
+    if isinstance(date, str):
+        try:
+            check_date = datetime.datetime.strptime(date, DATE_FORMAT)
+        except ValueError:
+            return False
+    else:
+        check_date = date
+    return now_year - valid_dates < check_date.year < now_year + valid_dates
+
+
+def is_event_text_valid(row: str) -> bool:
+    """Check if the row contains valid data"""
+    get_values = EVENT_PATTERN.findall(row)
```

Comment:
Prefer `.search` and check if `... is not None`

Reply:
Done 👍